### ISSUE TITLE
Should jwt-auth be linked with the built in auth of Laravel or not at all?

### ISSUE THREAD

**Issue by godbout** [tymondesigns/jwt-auth#94]
I'm having a hard time understanding how these auth systems work together.

I am currently using the register and login pages built in Laravel to send the data but the controllers receiving the data are ones I've made, and I use jwt-auth for authentication. I haven't been able to make that work yet as the attempt method always returns false. I assume it's because my users are saved with unencrypted passwords, which I don't know why. 

Should I get rid of all the built in auth or is the jwt-auth playing with it too? Can I use things like the password reset functionality or should I rewrite that? Is my User class need something special to now hash the password as it doesn't seem to be working out of the box (if someone can explain me why I'll be glad because I have absolutely no idea...).

So many questions :)

**Comment by godbout** [comment]
I was able to get something working by encrypting the password by adding a setPasswordAttribute method to my User class. Not sure why I need that now to encrypt the password as I've never had to add this before. Also, I don't understand how does the JWTAuth::attempt() method knows it needs to decrypt the password? Is it extending or using something auth thing from Laravel? OMG I'm so lost.

**Comment by tymondesigns** [comment]
@godbout Firstly, JWTAuth is not a drop in replacement for Laravel's Auth system - See https://github.com/tymondesigns/jwt-auth/issues/77#issuecomment-93020931

Also I think you may benefit from researching a little more about why and when you might use JWT's. Maybe look at a few blog posts around the web. 

Here is one that uses this exact package, and should give you more of an insight: http://www.toptal.com/web/cookie-free-authentication-with-json-web-tokens-an-example-in-laravel-and-angularjs

Thanks

**Comment by godbout** [comment]
I'm clear about a few things: the why and the when (I do APIs with Laravel and frontend with Angular (with no use of session), though I'm new to both) and also what is JWT. I've also checked many articles (like the one you provide) and many screencasts, but I'm lost in the magic of Laravel I think. 

So I was not sure if the default Auth controllers (AuthController and PasswordController) coming with Laravel were still useful. In theory I guess they are, but still not sure... With all I learnt, still no idea how this integrates. At the end I just did a "php artisan fresh" and all is gone, I'll try to figure out how all of this works later.

Thanks for your answer @tymondesigns
